homeassistant/components/lunatone/light.py

```diff
@@ -0,0 +1,112 @@
+"""Platform for Lunatone light integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from homeassistant.components.light import ColorMode, LightEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import LunatoneConfigEntry, LunatoneDevicesDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: LunatoneConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Lunatone Light platform."""
+    coordinator_info = config_entry.runtime_data.coordinator_info
+    coordinator_devices = config_entry.runtime_data.coordinator_devices
+
+    info_api = coordinator_info.info_api
+    devices_api = coordinator_devices.devices_api
+
+    # Add devices
+    async_add_entities(
+        [
+            LunatoneLight(coordinator_devices, device.id, info_api.serial_number)
+            for device in devices_api.devices
+        ],
+        update_before_add=True,
+    )
+
+
+class LunatoneLight(
+    CoordinatorEntity[LunatoneDevicesDataUpdateCoordinator], LightEntity
+):
+    """Representation of a Lunatone light."""
+
+    unique_id: str
+
+    _unavailable_logged: bool = False
+
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: LunatoneDevicesDataUpdateCoordinator,
+        device_id: int,
+        interface_serial_number: int,
+    ) -> None:
+        """Initialize a LunatoneLight."""
+        super().__init__(coordinator=coordinator, context=device_id)
```

Comment:
```suggestion
        super().__init__(coordinator=coordinator)
        self._device_id = device_id
```
lets store `self._device_id` in this class instead, since it makes it clearer than storing it in coordinator context

Reply:
Done with [7756571](https://github.com/home-assistant/core/pull/149182/commits/7756571c0a1c71d5b6eeb6ced8b217931855a2c9).